chatkit/src/main/java/com/pusher/chatkit/ChatManager.kt

```diff
@@ -94,20 +95,31 @@ class ChatManager @JvmOverloads constructor(
         tokenProvider = tokenProvider,
         tokenParams = null,
         logger = logger,
-        consumeEvent = {
-            if (it is CurrentUserReceived) {
-                currentUser = it.currentUser
-            }
-            consumer(it)
+        consumeEvent = { event ->
+            event.handleEvent()
+            consumer(event)
         }
     )
 
+    private fun ChatKitEvent.handleEvent() {
+        when (this) {
+            is CurrentUserReceived -> this@ChatManager.currentUser = currentUser
+            is ErrorOccurred -> logger.error(error.reason, error.asSystemError())
+            is RoomDeleted -> roomStore -= roomId
+            is RoomUpdated -> roomStore += room
+            is CurrentUserRemovedFromRoom -> currentUser?.id?.let { id -> roomStore[roomId]?.removeUser(id) }
```

Comment:
Is there a `CurrentUserAddedToRoom` (or similar) that's yet to be added here?

Reply:
Good spot :) And this is where the `plusAssign` will be used